contracts/modules/Exec.sol

```diff
@@ -100,6 +100,31 @@ contract Exec is BaseLogic {
         if (status == DEFERLIQUIDITY__DIRTY) checkLiquidity(account);
     }
 
+    /// @notice Defer liquidity checking for an array of accounts, to perform rebalancing, flash loans, etc. msg.sender must implement IDeferredLiquidityCheck
+    /// @param accounts The array of accounts to defer liquidity for
+    /// @param data Passed through to the onDeferredLiquidityCheck() callback, so contracts don't need to store transient data in storage
+    function deferLiquidityCheckExtended(address[] memory accounts, bytes memory data) external reentrantOK {
+        address msgSender = unpackTrailingParamMsgSender();
+
+        for (uint i = 0; i < accounts.length; ++i) {
+            address account = accounts[i];
+
+            require(accountLookup[account].deferLiquidityStatus == DEFERLIQUIDITY__NONE, "e/defer/reentrancy");
+            accountLookup[account].deferLiquidityStatus = DEFERLIQUIDITY__CLEAN;
+        }
+
+        IDeferredLiquidityCheck(msgSender).onDeferredLiquidityCheck(data);
+
+        for (uint i = 0; i < accounts.length; ++i) {
+            address account = accounts[i];
+
+            uint8 status = accountLookup[account].deferLiquidityStatus;
+            accountLookup[account].deferLiquidityStatus = DEFERLIQUIDITY__NONE;
+
+            if (status == DEFERLIQUIDITY__DIRTY) checkLiquidity(account);
+        }
+    }
```

Comment:
Looking at it here, I think it *would* make sense to re-use the code between the single and multi-versions by having the single call the multi. Technically it would cost a bit more gas, but I think it's worth it to keep the code clean.